src/App.tsx

```diff
@@ -1,14 +1,71 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { Todo } from './types/Todo';
+
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+
+export enum Filters {
+  all = 'all',
+  active = 'active',
+  completed = 'completed',
+}
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosFilter, setTodosFilter] = useState<Filters>(Filters.all);
+  const [query, setQuery] = useState('');
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  function getSelectedTodoById() {
+    return todos.find(todo => todo.id === selectedTodoId) || todos[0];
+  }
+
+  const handlerFilterTodos = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    event.preventDefault();
+
+    const newFilter = event.target.value as unknown as Filters;
+
+    return setTodosFilter(newFilter);
+  };
+
+  const handlerInputValue = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+
+    return setQuery(event.target.value);
+  };
+
+  const clearQuery = () => setQuery('');
+
+  const handleSelectTodo = (id: number) => setSelectedTodoId(id);
```

Comment:
this function wrapper is redundant. You can use original `filteredPeople` instead
FYI